### ISSUE TITLE
Getting the correct US image spacing for Clarius probes.

### ISSUE THREAD

**Issue by prash-p** [PlusToolkit/PlusLib#1124]
I'm using the nightly Clarius Cast build and am able to successfully stream images using PLUS.

However, since the US image dimensions depend on how the user has zoomed in on the Clarius Android/iOS app, the pixel dimensions are not fixed. My understanding is that the frame width x height is fixed (for example at the default 640 x 480 pixels). Is it possible to retrieve the actual pixel spacing of the image? I understand doing an image calibration and fixing the imaging depth can let you calculate the fixed image spacing, but this calibration is easy to void if the mobile app is accidentally touched during operation.

**Comment by Sunderlandkyl** [comment]
It looks like the device currently sends "micronsPerPixel" as a custom field. Can you see if it is being added to the image metatdata? In 3D Slicer, I think it would show up as a node attribute.

Ideally it should be sent as part of an ImageToTransducer transform, however that hasn't been implemented yet.

**Comment by prash-p** [comment]
Thanks - you do get a "miconsPerPixel" attribute that updates as you change the depth on the app:
![image](https://github.com/PlusToolkit/PlusLib/assets/5707552/566d52cf-1045-4080-aff6-d02e24c13a1a)

Follow up question - is it possible to read this attribute using the python OpenIGTLink library (pyigtl)?

**Comment by Sunderlandkyl** [comment]
Sure, check the metadata dictionary in the incoming message.
Probably something like: 
````
message.metadata["micronsPerPixel"]
````

**Comment by Sunderlandkyl** [comment]
Feel free to re-open if you have any issues!

**Comment by prash-p** [comment]
In pyigtl I have
```
message = client.wait_for_message("Image_Reference", timeout=30)
image = np.squeeze(message.image)
print(message.metadata)
```
But this just prints an empty dictionary with no keys/values. Any idea why the micronsPerPixel aren't getting sent? The same PLUS config file does sent this metadata as I can read it in slicer.

**Comment by Sunderlandkyl** [comment]
Right, Plus sends v1 header messages by default, which do not support metadata.
You can let Plus know that you can support v2 headers by sending a dummy message with header_version = 2.

````Python
dummyMessage = pyigtl.StringMessage("DummyMessage")
dummyMessage.header_version = 2
client.send_message(dummyMessage)
````

**Comment by prash-p** [comment]
Thanks - that fixed it!
